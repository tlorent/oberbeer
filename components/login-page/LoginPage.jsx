import React from 'react';

import './login-page.scss';
import { auth, provider } from '../../firebase';
import Search from '../search/Search';

class Login extends React.Component {
  state = {
    user: null
  };

  componentDidMount() {
    auth.onAuthStateChanged((user) => {
      if (user) {
        this.setState({ user });
      }
    });
  }

  login = () => {
    auth.signInWithPopup(provider).then(({ user }) => {
      this.setState({
        user
      });
    });
  };

  logout = () => {
    auth.signOut().then(() => this.setState({ user: null }));
  };

  render() {
    const { user } = this.state;
    return (
      <React.Fragment>
        <div className="login">
          {user ? (
            <button className="login__button" type="button" onClick={() => this.logout()}>
              Log Out
            </button>
          ) : (
            <button className="logout__button" type="button" onClick={() => this.login()}>
              Log In With Google
            </button>
          )}
        </div>
        {user && <Search />}
      </React.Fragment>
    );
  }
}

export default Login;
