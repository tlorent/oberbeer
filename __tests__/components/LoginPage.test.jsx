import React from 'react';
import { shallow } from 'enzyme';

import LoginPage from '../../components/login-page/LoginPage';

test('should render LoginPage correctly', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper.find('button').text()).toBe('Log In');

  expect(wrapper).toMatchSnapshot();
});
