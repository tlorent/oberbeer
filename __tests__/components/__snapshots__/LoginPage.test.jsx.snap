// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<Fragment>
  <button
    className="login__button"
    type="button"
  >
    Log In
  </button>
</Fragment>
`;
